pandora-client-web/src/components/settings/accountSettings.tsx

```diff
@@ -198,3 +200,43 @@ function LabelColor({ account }: { account: IDirectoryAccountInfo; }): ReactElem
 		</fieldset>
 	);
 }
+
+function DisplayName({ account }: { account: IDirectoryAccountInfo; }): ReactElement {
+	const directory = useDirectoryConnector();
+	const nextAllowedChange = account.settingsCooldowns.displayName ?? 0;
+	const current = account.settings.displayName ?? account.username;
+	const [name, setName] = useState(current);
+
+	const onSetDisplayName = useEvent(() => {
+		if (name === current)
+			return;
+		if (nextAllowedChange && nextAllowedChange > Date.now()) {
+			toast(`You can change your display name again in ${FormatTimeInterval(nextAllowedChange - Date.now())}`, TOAST_OPTIONS_ERROR);
+			return;
+		}
+		const displayName = account.username === current ? null : current;
```

Comment:
This should use the new name, not the old one...
```suggestion
		const displayName = name === current ? null : name;
```